modules/plugin/jdbc/jdbc-sqlserver/src/main/java/org/geotools/data/sqlserver/SQLServerDialect.java

```diff
@@ -165,6 +166,9 @@ public void registerSqlTypeNameToClassMappings(Map<String, Class<?>> mappings) {
         mappings.put("uniqueidentifier", UUID.class);
         mappings.put("time", Time.class);
         mappings.put("date", Date.class);
+        mappings.put("datetime", Timestamp.class);
+        mappings.put("datetime2", Timestamp.class);
+        mappings.put("datetimeoffset", Timestamp.class);
```

Comment:
This conversion looses the timezone offset, not sure if that is what you expect to happen

Reply:
Thanks, didn't consider that. I will look into it.

Reply:
I've switched so it maps to driver specific type `microsoft.sql.DateTimeOffset` to ensure proper serialization. I've also added more test details. Does it seem sensible to you now @mprins?

Reply:
@mprins will follow up too, but sharing my 2 cents... DateTimeOffset ensure no semantic is lost, but at the same time, it's not part of the "java.util.Date" inheritance hierarchy, so the rest of GeoTools (and GeoServer output formats) might not know what to do with it, it might toString it if you're lucky, or throw an error due to unrecognized type.

Ideally it could be mapped to one of Java 8 date primitives, but we'd be basically back to square one again, as the rest of GeoTools also does not know how to use them.

I see @ianturton added a JTDSDateConverterFactory to handle the JTDS driver dates, @ianturton would a dedicated converter form DateTimeOffset help in this case, and how is the wiring done?

Reply:
Well.. at least I can confirm it works well as WFS datastore as used in GeoServer. I carry this patch as hotfix for several GeoServer deployments. The behaviour without this patch is that the columns are silently omitted in GeoServer which IMHO is worse than crashing.

Reply:
With ``java.sql.Timestamp`` it's going to work but you might lose information (which could also be handled via documentation). 
Did you test it with ``DateTimeOffset`` instead? I assume the patch you have been carrying around for several deployments was the original one you submitted, using Timestamp only... or not?

Reply:
The initial patch worked but lost timezone. The latest patch as is in this PR also works and retains timezone in WFS output.

Reply:
Noticed some bad copy paste for the testInsert method which I've now fixed.

Reply:
Found that there is an official `java.time.OffsetDateTime` that seems to work as well as the microsoft specific type. Switching to it.

Reply:
`java.time.OffsetDateTime` does not inherit from `java.util.Date` so it won't help for the potential issues mentioned by @aaime but I would think it's the more appropriate and portable choice here.

Reply:
I agree with @aaime; using a type (`java.time.OffsetDateTime`) that does not cast to `java.util.Date` will probably cause more grief than getting results that are a timezone worth of offset off; I seem to recall that can be fixed using a connection parameter, not sure about that.

As I said before I would be fine adding `datetimeoffset` as a `Timestamp.class` and live with the consequences. 

FWIW postgis uses `java.sql.Timestamp` for timezoned `timestamptz` data as well
https://github.com/geotools/geotools/blob/4b968cfe0559d131c24cffa841122442fe51c466/modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostGISDialect.java#L1109

ps. the Oracle driver does some fancy conversion of oracle time stamps with a converter too, as the oracle things cannot be cast to java.util classes

Reply:
if `java.time.OffsetDateTime` works for your WFS's I'm happy with that

Reply:
I'm not happy to revert this and lose the timezone again, because it's actually providing the correct behavior at the end user here now.

Does this mean I have to provide "fancy conversion" somehow in this PR to get it accepted? Or even possibly a complete migration to java.time.* (but I suppose that is a pretty large undertaking...)?

Reply:
Verified working end to end GeoServer WFS also after switching to `java.time.OffsetDateTime`.

Reply:
@aaime considering the above is this acceptable as is to you?